roles/repositories/tasks/main.yml

```diff
@@ -0,0 +1,50 @@
+---
+- name: 'Enable Red Hat Repositories'
+  theforeman.foreman.repository_set:
+    username: "{{ username }}"
+    password: "{{ password }}"
+    server_url: "{{ server_url }}"
+    validate_certs: "{{ validate_certs | default(omit) }}"
+    organization: "{{ organization }}"
+    product: "{{ item.0.name }}"
+    name: "{{ item.1.name }}"
+    state: enabled
+    repositories:
```

Comment:
the repo set module has a `all_repositories` parameter: https://theforeman.github.io/foreman-ansible-modules/develop/plugins/repository_set_module.html#parameter-all_repositories

Should we expose that here? There is also an example how to combine that with `repositories` so that the user can have a single data structure but some sets have `repositories` and the others end up being `all`.

Reply:
A caveat is that `all_repositories` requires either `name` + `product`, OR `label` to be specified. I added support for the labels but not for product + name. I could add that still if its needed but it would require a 3rd task or a bit of rethinking my approach to the looping.

Reply:
I also tried to test multiple labels but using the `rhel-server-rhscl-6-rpms` label resulted in working Ansible but "The path /content/dist/rhel/server/6/6.9/i386/rhscl/1/os does not seem to be a valid repository. If you think this is an error, please try refreshing your manifest." from Katello